### ISSUE TITLE
Feature - newlines around multiline expressions

### ISSUE THREAD

**Issue by Epigene** [rubocop-hq/rubocop#6444]
Hi,

Perhaps such a cop already exists and I've missed it, but I'd like to enforce newlines around all __multi-line__ expressions such that expression beginnings and expression ends are separated by a newline.

An example:

```rb
# bad
def method
  puts(
    "long line"
  )  
  1 + 1
  true
  loop do
  end
end

# good
1. def method
2.  puts(
3.    "long line"
4.  )
5. 
6.  1 + 1
7.  true
8.
9.  loop do
10.  end
11. end
```

Key features:
  1. Consecutive "beginnings" and "ends" are not separated by a newline (lines 1.-2., 10.-11.)
  2. "Beginning" and "end" types need not match - a multiline method call gets separated from a oneliner addition beginning, and oneliner addition end gets separated from multiline block beginning (lines 4.-6. and 7.-9.)
  3. A single statement's own beginning and end do not get seperated (lines 9.-10.)
  4. Oneliner statements are not separated (lines 6.-7.)
